runtime/drivers/bigquery/sql_store.go

```diff
@@ -43,41 +44,93 @@ func (c *Connection) QueryAsFiles(ctx context.Context, props map[string]any, opt
 		return nil, err
 	}
 
-	client, err := bigquery.NewClient(ctx, srcProps.ProjectID, opts...)
-	if err != nil {
-		if strings.Contains(err.Error(), "unable to detect projectID") {
-			return nil, fmt.Errorf("projectID not detected in credentials. Please set `project_id` in source yaml")
+	now := time.Now()
+
+	var client *bigquery.Client
+	var it *bigquery.RowIterator
+	bqClientFn := func(projectID string) (*bigquery.Client, error) {
+		client, err := bigquery.NewClient(ctx, projectID, opts...)
+		if err != nil {
+			if strings.Contains(err.Error(), "unable to detect projectID") {
+				return nil, fmt.Errorf("projectID not detected in credentials. Please set `project_id` in source yaml")
+			}
+			return nil, fmt.Errorf("failed to create bigquery client: %w", err)
 		}
-		return nil, fmt.Errorf("failed to create bigquery client: %w", err)
+		return client, nil
 	}
 
-	if err := client.EnableStorageReadClient(ctx, opts...); err != nil {
-		client.Close()
-		return nil, err
-	}
+	r := regexp.MustCompile("(?i)^\\s*SELECT\\s+\\*\\s+FROM\\s+(`?[a-zA-Z0-9_.-]+`?)\\s*$")
```

Comment:
This is an extension of the current approach. If there were two forms for loading data from BQ, we would be anyway needed to somehow direct the user to choose the right one for their needs.

> This regex variable can be global ?

It might be unified to be global but as of now, backticks are specific for BQ

Reply:
> It might be unified to be global but as of now, backticks are specific for BQ

Sorry I meant global within this pkg. Since this is a fixed regex and regex compilation can be computationally expensive we can avoid doing this for every query.